### ISSUE TITLE
Segmentation Fault when using find() or creating instance of model

### ISSUE THREAD

**Issue by iamdann** [phalcon/incubator#887]
I'm trying to setup MongoDB using this package and it's ODM. Currently everything seems to work except the instantiation of models extending the `\Phalcon\Mvc\MongoCollection` class. 

My `services.php`: 
```
use Phalcon\Db\Adapter\MongoDB\Client as MongoClient;
use Phalcon\Mvc\Collection\Manager;

//MongoDB Database
$di->set('mongo', function () {
    $mongo = new MongoClient('mongodb://' . getenv('MONGODB_HOST') . ':' . getenv('MONGODB_PORT'));

    return $mongo->selectDatabase(getenv('MONGODB_NAME'));
});

//Collection Manager
$di->set('collectionManager', function () {
    return new Manager();
});
```

My `Pages.php` model: 
```
class Pages extends \Phalcon\Mvc\MongoCollection
{
    public function getSource(){
        return 'pages';
    }
}
```

When I try to do the following, my page results in a `The connection was reset` error and `child pid <id> exit signal Segmentation fault (11)` notice in the Apache logs: 
```
$pages = new Pages();
```

OR

```
Pages::find();
```

However, I can do `$mongo->listDatabases()` and get the following result: 
```
object(Phalcon\Db\Adapter\MongoDB\Model\DatabaseInfoLegacyIterator)#167 (1) { ["databases":"Phalcon\Db\Adapter\MongoDB\Model\DatabaseInfoLegacyIterator":private]=> array(2) { [0]=> array(3) { ["name"]=> string(5) "admin" ["sizeOnDisk"]=> float(49152) ["empty"]=> bool(false) } [1]=> array(3) { ["name"]=> string(5) "local" ["sizeOnDisk"]=> float(32768) ["empty"]=> bool(false) } } } 
```

The [MongoDB PHP Driver](https://secure.php.net/manual/en/set.mongodb.php) also seems to work: 
```
$collection = (new Client('mongodb://' . getenv('MONGODB_HOST') . ':' . getenv('MONGODB_PORT')))->danqcms->pages;
$collection->find()->toArray();
```

### Details

* Phalcon Framework version: 3.4.2
* Phalcon Incubator version: 3.4
* PHP Version: 7.3.1
* Operating System: MacOS, Docker Container running Debian
* Server: Apache

**Comment by ruudboon** [comment]
Phalcon 3.4.2 isn't compatible with PHP 7.3. 
V4 for is but still in alpha and continues mongo support is still part of a discussion [here](https://github.com/phalcon/cphalcon/issues/13697)

**Comment by iamdann** [comment]
Wasn't quite sure if my current setup was supported due to the message on the top of the page found [here](https://docs.phalconphp.com/3.2/en/db-odm). Thanks.

**Comment by lubber-de** [comment]
Phalcon 3.4.3 is now available for PHP 7.3, so this might get re-opened again
